### PR TITLE
Ask the users if they want to check for updates

### PR DESCRIPTION
**Pull request by iTrooz**
The principle is that the default value is 2 ("not set") and the popup makes it switch to 1 ("check for updates") or 0 ("do not check for updates") (0 being the default is the popup is closed)

I wasn't sure how to do something cleaner

### CODE REVIEW

**Comment by iTrooz** [comment]
https://github.com/WerWolv/ImHex/issues/794

File: main/source/init/tasks.cpp [hunk 1]
```diff
@@ -25,23 +25,26 @@ namespace hex::init {
     using namespace std::literals::string_literals;
 
     static bool checkForUpdates() {
-        hex::Net net;
+        int showCheckForUpdates = ContentRegistry::Settings::read("hex.builtin.setting.general", "hex.builtin.setting.general.check_for_updates", 2);
```
Comment (WerWolv) [inline main/source/init/tasks.cpp:28]: Could you document in a comment what the different values for this setting mean? 0, 1 and 2 alone aren't really self explanatory

File: main/source/init/tasks.cpp [hunk 2]
```diff
@@ -25,23 +25,26 @@ namespace hex::init {
...
+        if(showCheckForUpdates==1){
```
Comment (WerWolv) [inline main/source/init/tasks.cpp:29]: Spaces before and after operator are missing

File: plugins/builtin/source/content/settings_entries.cpp [hunk 3]
```diff
@@ -35,6 +35,17 @@ namespace hex::plugin::builtin {
 
         /* General */
 
+        ContentRegistry::Settings::add("hex.builtin.setting.general", "hex.builtin.setting.general.check_for_updates", 2, [](auto name, nlohmann::json &setting) {
```
Comment (WerWolv) [inline plugins/builtin/source/content/settings_entries.cpp:38]: 1 and 2 will both be treated here as "Enabled". Is that the desired effect?

File: plugins/builtin/source/content/welcome_screen.cpp [hunk 4]
```diff
@@ -517,6 +517,17 @@ namespace hex::plugin::builtin {
                 loadDefaultLayout();
         });
 
+        EventManager::subscribe<EventWindowInitialized>([]{
+            int showCheckForUpdates = ContentRegistry::Settings::read("hex.builtin.setting.general", "hex.builtin.setting.general.check_for_updates", 2);
+            if(showCheckForUpdates==2){
```
Comment (WerWolv) [inline plugins/builtin/source/content/welcome_screen.cpp:522]: Spaces before and after operator are missing

File: plugins/builtin/source/content/welcome_screen.cpp [hunk 5]
```diff
@@ -517,6 +517,17 @@ namespace hex::plugin::builtin {
...
+                ContentRegistry::Settings::write("hex.builtin.setting.general", "hex.builtin.setting.general.check_for_updates", 0); 
+                View::showYesNoQuestionPopup("hex.builtin.welcome.check_for_updates_text"_lang, [] {
```
Comment (WerWolv) [inline plugins/builtin/source/content/welcome_screen.cpp:524]: Formatting of the lambdas here is kinda weird. I'd put the head of the lambda and its content on two different lines and also move the head of the first lambda on a new line

File: plugins/builtin/source/lang/en_US.cpp [hunk 6]
```diff
@@ -45,6 +45,9 @@ namespace hex::plugin::builtin {
                     { "hex.builtin.welcome.learn.plugins.link", "https://github.com/WerWolv/ImHex/wiki/Plugins-Development-Guide" },
                 { "hex.builtin.welcome.header.various", "Various" },
                 { "hex.builtin.welcome.tip_of_the_day", "Tip of the Day" },
+                { "hex.builtin.welcome.check_for_updates", "Check for updates" },
```
Comment (WerWolv) [inline plugins/builtin/source/lang/en_US.cpp:48]: These should also be added to the other language files

**Review by WerWolv — CHANGES_REQUESTED** [review]

**Comment by WerWolv** [comment]
Thank you!

File: main/source/init/tasks.cpp (hunk 1 above)
Comment (iTrooz) [reply main/source/init/tasks.cpp:28]: Sure ! Actually, aren't enums just named integers in C++ ? In this case, may I create an enum somewhere ? (I'm not sure where that would fit)

File: main/source/init/tasks.cpp (hunk 2 above)
Comment (iTrooz) [reply main/source/init/tasks.cpp:29]: Oops, sorry for making you repeat yourself in this PR, I should have checked

File: plugins/builtin/source/content/welcome_screen.cpp (hunk 5 above)
Comment (iTrooz) [reply plugins/builtin/source/content/welcome_screen.cpp]: Like that ? (see my last commit)

File: main/source/init/tasks.cpp (hunk 1 above)
Comment (WerWolv) [reply main/source/init/tasks.cpp:28]: Yeah an enum would be great for this too. Just requires some casting when writing it to the json object probably

**Review by WerWolv — APPROVED** [review]

File: plugins/builtin/source/content/welcome_screen.cpp (hunk 5 above)
Comment (WerWolv) [reply plugins/builtin/source/content/welcome_screen.cpp]: Yeah an enum would be great for this too. Just requires some casting when writing it to the json object probably

Comment (iTrooz) [reply plugins/builtin/source/content/welcome_screen.cpp]: Where should I create the enum ? We would need it in `main` and `builtin`, so how about in `tasks.hpp` ?

Comment (WerWolv) [reply plugins/builtin/source/content/welcome_screen.cpp]: Right it's split up between multiple files...
Yeah I honestly also don't know where to put it best.
For now I'd probably just put a comment above the definition of the setting.

Comment (iTrooz) [reply plugins/builtin/source/content/welcome_screen.cpp]: Done !

**Review by WerWolv — APPROVED** [review]

**Comment by WerWolv** [comment]
Thank you!

**Pull request merged**